src/ctap/mod.rs

```diff
@@ -428,37 +434,55 @@ where
             return Err(Ctap2StatusCode::CTAP2_ERR_UNSUPPORTED_ALGORITHM);
         }
 
-        let use_hmac_extension =
-            extensions.map_or(Ok(false), |e| e.has_make_credential_hmac_secret())?;
-        if use_hmac_extension && !options.rk {
-            // The extension is actually supported, but we need resident keys.
-            return Err(Ctap2StatusCode::CTAP2_ERR_UNSUPPORTED_EXTENSION);
-        }
+        let (use_hmac_extension, cred_protect_policy) = if let Some(extensions) = extensions {
+            let mut cred_protect = extensions
+                .make_credential_cred_protect_policy()
+                .transpose()?;
+            if cred_protect
+                .as_ref()
+                .unwrap_or(&CredentialProtectionPolicy::UserVerificationOptional)
```

Comment:
Here if the enum type is `Copy`, you don't need the `as_ref` part, nor the reference inside the `unwrap_or`. This will be more readable.

Reply:
Done, thanks.